src/hooks/useOpenLoanHandler.tsx

```diff
@@ -458,6 +467,98 @@ export const useOpenLoanHandler = () => {
 
   setValue('collateralType', collateralType)
 
+  const approveTransaction = async () => {
+    if (collateralType.type === 'Token A' && !collateralAmt0Str) {
+      notifyError('Please enter a valid amount')
+      return
+    }
+
+    if (collateralType.type === 'Token B' && !collateralAmt1Str) {
+      notifyError('Please enter a valid amount')
+      return
+    }
+
+    if (collateralType.type === 'Both Tokens' && (!collateralAmt0Str || !collateralAmt1Str)) {
+      notifyError('Please enter a valid amount')
+      return
+    }
+
+    if (collateralType.type === 'Liquidity Pool Tokens' && !collateralAmt0Str) {
+      notifyError('Please enter a valid amount')
+      return
+    }
+
+    if (accountInfo?.address && provider) {
+      let approvalA, approvalB, approvalCFMM
+      if (collateralType.type === 'Token A' || collateralType.type === 'Both Tokens') {
+        let token0Contract = new ethers.Contract(
+          token0.address,
+          IERC20.abi,
+          accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
+        )
+
+        approvalA = await validateAllowance(
+          accountInfo.address,
+          token0Contract,
+          BigNumber.from(collateralAmt0Str),
+          POSITION_MANAGER_ADDRESS || ''
+        )
+
+        if (!approvalA) {
+          return
```

Comment:
need to setIsApproved(false)?

Reply:
Default value is always false

Reply:
but what if they decide to make changes, and it's not valid anymore but the approve is enabled?

Reply:
Done. Now when data changes after the approval it will be false